### ISSUE TITLE
Instructor copying sessions: preserve time when auto-changing dates

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#12045]
v8.21

Current: When an instructor is copying a session to a new course, dates are auto-changed if they don't fall within the allowed time period for new sessions. e.g.,
![image](https://user-images.githubusercontent.com/1673303/215253145-b1035a34-79e6-4c97-8b0d-47e986ab47ad.png)

Suggestion: Preserve the original time and change the date only, as it is most likely the instructor will want to keep the same times as before.

**Comment by domlimm** [comment]
hello prof, may i pick up this issue?

**Comment by damithc** [comment]
Sure. @xinyee20 any advice for @domlimm

**Comment by xyliew25** [comment]
@damithc not sure if you're referring to me but should such adjustments be applied to sessions copied over when an instructor is copying a course?

**Comment by damithc** [comment]
> @damithc not sure if you're referring to me but should such adjustments be applied to sessions copied over when an instructor is copying a course?

Yes, @xyliew25 (sorry, tagged the wrong person). Yes, it's good to preserve session times when copying a course as well.

**Comment by damithc** [comment]
Although not as important, I think preserving the day of the week has some value too. At least in my use case I always set the start/end on the same day of the week (e.g., Friday 2pm) and it is nice that the auto-set date already reminds me of the date I usually use.

**Comment by domlimm** [comment]
@andreacastellanos hello! i understand that i have not opened a PR yet, and it's not exactly clear if i'm assigned to this but appreciate it if you could please check the conversation in the issue before creating a PR in the future or even ask. thanks! :)

**Comment by domlimm** [comment]
Hello prof, could I pick up this issue from where Andre left off, and credit him for his part too?

**Comment by samuelfangjw** [comment]
Sure @domlimm.

**Comment by andremralves** [comment]
> @andremralves hello! i understand that i have not opened a PR yet, and it's not exactly clear if i'm assigned to this but appreciate it if you could please check the conversation in the issue before creating a PR in the future or even ask. thanks! :)

Hi @domlimm, sorry about that. I was working on a task from my university that was related to solving an issue from this repository and describing the steps during the process. As I saw it was about one week that no one was talking about this issue I  opened a PR. I wasn't with much time and I ended up forgetting to ask before I started.

For me you have total right to continue this issue. I'm sorry again.
